src/MultiFormats/MultiAddress.cs

```diff
@@ -0,0 +1,565 @@
+        ///   to a human.
+        /// </summary>
+        public override string ToString()
+        {
+            using (var s = new StringWriter())
+            {
+                Write(s);
+                return s.ToString();
+            }
+        }
+
+        /// <summary>
+        ///   Returns the IPFS binary representation as a byte array.
+        /// </summary>
+        /// <returns>
+        ///   A byte array.
+        /// </returns>
+        /// <remarks>
+        ///   The binary representation is a sequence of <see cref="NetworkProtocol">network protocols</see>.
+        /// </remarks>
+        public byte[] ToArray()
+        {
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                return ms.ToArray();
+            }
+        }
+
+        /// <summary>
+        ///   Implicit casting of a <see cref="string"/> to a <see cref="MultiAddress"/>.
+        /// </summary>
+        /// <param name="s">The string representation of a <see cref="MultiAddress"/>.</param>
+        /// <returns>A new <see cref="MultiAddress"/>.</returns>
+        public static implicit operator MultiAddress(string s) { return new MultiAddress(s); }
+
+        /// <summary>
+        ///   Try to create a <see cref="MultiAddress"/> from the specified
+        ///   string.
+        /// </summary>
+        /// <param name="s">
+        ///   The string representation of a multi address, such as "/ip4/1270.0.01/tcp/5001".
+        /// </param>
+        /// <returns>
+        ///   <b>null</b> if the string cannot be parsed; otherwise a <see cref="MultiAddress"/>.
+        /// </returns>
+        public static MultiAddress TryCreate(string s)
+        {
+            try
+            {
+                return new MultiAddress(s);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        ///   Try to create a <see cref="MultiAddress"/> from the specified
+        ///   the binary encoding.
+        /// </summary>
+        /// <param name="bytes">
+        ///   The binary encoding of a multiaddress.
+        /// </param>
+        /// <returns>
+        ///   <b>null</b> if the bytes cannot be parsed; otherwise a <see cref="MultiAddress"/>.
+        /// </returns>
+        public static MultiAddress TryCreate(byte[] bytes)
+        {
+            try
+            {
+                return new MultiAddress(bytes);
+            }
+            catch
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        ///   Conversion of a <see cref="MultiAddress"/> to and from JSON.
+        /// </summary>
+        /// <remarks>
+        ///   The JSON is just a single string value.
+        /// </remarks>
+        private sealed class Json : JsonConverter
+        {
+            public override bool CanConvert(Type objectType) { return true; }
+
+            public override bool CanRead => true;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private property 'CanRead'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979568)